### PR TITLE
86cuex0z1/UI/Trip details

### PR DESCRIPTION
**Pull request by seraGeorge**
### Summary:
This pull request introduces trip details section of the accordian, including the creation of custom alert and long button components and the incorporation of seat legend and layout along with the total amount incurred. 

### Changes:

**1. Custom Components:**
- Added `StyledAlert` component with custom styles and props.
- Created `StyledButton` component for a distinctive visual style.

**2. Seat Display:**
- Incorporated a seat legend and seat layout to enhance the user experience.
- Incorporated a delete selection button for resetting the selection.

**3. Translation:**
- Added translation in English and Spanish using i18n.

**4. Styling and Modularization:**
- Implemented styled components for all components.
- Modularized the code for improved maintainability.

### Additional Notes:
Please review the changes and provide feedback on the implemented components, folder structures, UI, and any suggestions for improvements or identified issues.

### CODE REVIEW

**Comment by akshay-krishna** [comment]
Add changelog

File: src/components/ui/TripCard/AccordianDetails/Checkout/Checkout.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                    <Typography variant="body2" className="value">
+                        {totalFareInString}
+                    </Typography>
+                </Stack>
+            )}
+            <StyledButton
+                variant="contained"
+                disabled={!(noOfSeats > 0 && farePerSeat > 0)}
+            >
+                Book now
```
Comment (sania-salim) [inline src/components/ui/TripCard/AccordianDetails/Checkout/Checkout.tsx]: Missed out moving "Book Now" to tripDetails in language support

File: src/components/ui/TripCard/AccordianDetails/SeatLegend/SeatLegend.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import { Typography } from '@mui/material';
+import Seat from '../../../../SeatLayout/Seat';
+import SeatLegendWrapper from './SeatLegend.styled';
+import { useTranslation } from 'react-i18next';
+
+export const SeatLegend = () => {
```
Comment (sania-salim) [inline src/components/ui/TripCard/AccordianDetails/SeatLegend/SeatLegend.tsx]: Could make the size of legend smaller than actual seat size, especially the selected color stand out in contrast.

File: src/components/ui/TripCard/TripCardAccordion.tsx [hunk 3]
```diff
@@ -9,6 +7,7 @@ import seatIcon from '../../../assets/SeatIcon.svg';
 import sleeperIcon from '../../../assets/SleeperIcon.svg';
 import Stack from '@mui/material/Stack';
 import { TripAccordionStyled } from './TripCard.styled';
+import { TripCardDetails } from './AccordianDetails/TripCardDetails';
 
 interface ITripCardAccordion {
```
Comment (sania-salim) [inline src/components/ui/TripCard/TripCardAccordion.tsx:12]: Please correct the spelling "Accordian" to "Accordion" in AccordianDetails folder, assets and if present elsewhere.

File: src/components/ui/TripCard/AccordianDetails/DetailsGrid/DetailsGrid.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            imageAlt: 'Timer',
+        },
+        {
+            title: t('busType'),
+            value: 'AC, Seater',
+            imageSrc: bus,
+            imageAlt: 'Bus',
+        },
+    ];
+
```
Comment (sania-salim) [inline src/components/ui/TripCard/AccordianDetails/DetailsGrid/DetailsGrid.tsx]: Incorporate changing icon color when switching to dark theme

File: src/components/ui/TripCard/AccordianDetails/DetailsGrid/DetailItem/DetailItem.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    imageAlt: string;
+}
+export const DetailsItem = ({
+    title,
+    value,
+    imageSrc,
+    imageAlt,
+}: IDetailsItemProps) => {
+    return (
+        <DetailsItemWrapper direction={'row'} justifyContent={'space-between'}>
```
Comment (sania-salim) [inline src/components/ui/TripCard/AccordianDetails/DetailsGrid/DetailItem/DetailItem.tsx]: Suggesting that the content be left aligned for readability.

File: src/components/ui/TripCard/AccordianDetails/TripCardDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    };
+    const clearSelectedSeats = () => {
+        setSelectedSeats([]);
+    };
+
+    const { t } = useTranslation('tripDetails');
+
+    return (
+        <AccordionDetails>
+            <Stack direction={'column'} p={3} pt={0} spacing={4}>
```
Comment (sania-salim) [inline src/components/ui/TripCard/AccordianDetails/TripCardDetails.tsx]: Provide more white space within the accordion by adding padding between accordion title and content and bus layout.

**Review by sania-salim — CHANGES_REQUESTED** [review]

File: src/components/ui/TripCard/AccordionDetails/DetailsGrid/DetailsGrid.tsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+            <Grid container rowSpacing={2} columnSpacing={30}>
+                {detailsData.map((detail, index) => (
+                    <Grid item key={index} xs={12} sm={12} md={6} lg={6}>
+                        {/* 
+                        For small and medium screens (xs, sm), it will be a single column.
+                        For screens equal to or larger than md (medium) or lg(large), it will be 2 columns.*/}
+                        <DetailsItem
+                            title={detail.title}
+                            value={detail.value}
+                            imageSrc={detail.imageSrc}
```
Comment (visakhvqburst) [inline src/components/ui/TripCard/AccordionDetails/DetailsGrid/DetailsGrid.tsx:50]: Do we have any scenario where the image src URL will get as empty? what will be shown in the UI at the time?

File: src/components/ui/TripCard/AccordionDetails/DetailsGrid/DetailItem/DetailItem.tsx [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+export const DetailsItem = ({
+    title,
+    value,
+    imageSrc,
+    imageAlt,
+}: IDetailsItemProps) => {
+    return (
+        <DetailsItemWrapper direction={'row'} spacing={3}>
+            <Stack direction={'row'} spacing={1}>
+                <img src={imageSrc} alt={imageAlt} />
```
Comment (visakhvqburst) [inline src/components/ui/TripCard/AccordionDetails/DetailsGrid/DetailItem/DetailItem.tsx]: Do we have any scenario where the image src URL will get as empty? what will be shown in the UI at the time?

File: src/components/ui/TripCard/AccordionDetails/TripCardDetails.tsx [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+                    </Stack>
+                    <SeatLayout
+                        layoutName={layoutNames.volvo25}
+                        seats={seats}
+                        selectedSeats={selectedSeats}
+                        updateSelectedSeats={updateSelectedSeats}
+                    />
+                </Stack>
+
+                <Checkout noOfSeats={selectedSeats.length} farePerSeat={1200} />
```
Comment (visakhvqburst) [inline src/components/ui/TripCard/AccordionDetails/TripCardDetails.tsx]: Why don't we define a constant instead of a hardcoded value?

**Comment by seraGeorge** [comment]
> Add changelog

Resolved in 2debb530cc7a9ce9301cb6a77ba79b7fea3e9261

File: src/components/ui/TripCard/AccordianDetails/Checkout/Checkout.tsx (hunk 1 above)
Comment (seraGeorge) [reply src/components/ui/TripCard/AccordianDetails/Checkout/Checkout.tsx]: Resolved in 803f240d729e1dd4afe20dfe0a735440bd3ece49

File: src/components/ui/TripCard/AccordianDetails/SeatLegend/SeatLegend.tsx (hunk 2 above)
Comment (seraGeorge) [reply src/components/ui/TripCard/AccordianDetails/SeatLegend/SeatLegend.tsx]: Resolved in 4a41864f32f24fb644a205d535062fd54480087c

File: src/components/ui/TripCard/TripCardAccordion.tsx (hunk 3 above)
Comment (seraGeorge) [reply src/components/ui/TripCard/TripCardAccordion.tsx:12]: Resolved in 717b41915f078c22341fc331491f0017361434f0

File: src/components/ui/TripCard/AccordianDetails/DetailsGrid/DetailsGrid.tsx (hunk 4 above)
Comment (seraGeorge) [reply src/components/ui/TripCard/AccordianDetails/DetailsGrid/DetailsGrid.tsx]: Resolved in b83afc0b27039cc8831d18f2be4df5ba393562e8

File: src/components/ui/TripCard/AccordianDetails/DetailsGrid/DetailItem/DetailItem.tsx (hunk 5 above)
Comment (seraGeorge) [reply src/components/ui/TripCard/AccordianDetails/DetailsGrid/DetailItem/DetailItem.tsx]: Resolved in 209caa9248812fc845de2ab97206e85fa7cbece5

File: src/components/ui/TripCard/AccordianDetails/TripCardDetails.tsx (hunk 6 above)
Comment (seraGeorge) [reply src/components/ui/TripCard/AccordianDetails/TripCardDetails.tsx]: Resolved in 5137345a6671304eb7b016e929f5bec212e570c6

File: src/components/ui/TripCard/AccordionDetails/DetailsGrid/DetailItem/DetailItem.tsx (hunk 8 above)
Comment (seraGeorge) [reply src/components/ui/TripCard/AccordionDetails/DetailsGrid/DetailItem/DetailItem.tsx]: The image source URL doesn't become empty, as the elements are fixed and won't change. These are part of the layout.

File: src/components/ui/TripCard/AccordionDetails/TripCardDetails.tsx (hunk 9 above)
Comment (seraGeorge) [reply src/components/ui/TripCard/AccordionDetails/TripCardDetails.tsx]: Resolved in 308e5f2b802bf6010c03123b1913c1c7d500ee91

**Review by sania-salim — APPROVED** [review]

**Review by akshay-krishna — APPROVED** [review]

**Pull request merged**